<launch>
    <arg name="Joffsets" default="[60, 60, 5, 100, 60, 90]"/>

    <node name="grasp_key_input" pkg="sitl_dvrk" type="grasp_key_input.py" output="screen">
    </node>

    <node name="grasp_sample" pkg="sitl_dvrk" type="grasp_tissue_v2.py" output="screen">
        <param name="arm_name"           type="str"       value="PSM2"/>
        <param name="expected_interval"  type="double"    value="0.01"/>
        <param name="cnt_msg"            type="str"       value="/seg/sample/cnt_3d"/>
        <param name="bnd_msg"            type="str"       value="/seg/sample/bnd_3d"/>
        <param name="ctrd_msg"           type="str"       value="/seg/sample/ctrd_3d"/>
        <param name="tf_path"            type="str"       value="/home/$(env USER)/aruco_data/base_tfs.yaml"/>
        <param name="calib_fn"           type="str"       value="/home/$(env USER)/aruco_data/psm2_calib_results_final_new_v2.mat"/>
        <!-- <param name="pedal_flag"         type="bool"      value="true"/> -->
        <param name="Tweight"            type="double"    value="1.0"/>
        <param name="Rweight"            type="double"    value="0.2"/>
        <param name="align_dist"         type="double"    value="0.03"/>
        <!-- <param name="dist2afgrasp"       type="double"    value="0.03"/> -->
        <param name="grasp_dist"         type="double"    value="0.02"/>
        <param name="window_size"        type="int"       value="15"/>
        <param name="window_thr"         type="double"    value="3e-4"/>
        <rosparam param="Joffsets" subst_value="True"> $(arg Joffsets) </rosparam>
    </node>

    <?ignore

    ?>
</launch>